DigitalLearningSolutions.Web/Views/TrackingSystem/CentreConfiguration/RegistrationPrompts/Index.cshtml

```diff
@@ -8,100 +8,48 @@
+      <p class="nhsuk-breadcrumb__back">
+        <a class="nhsuk-breadcrumb__backlink" asp-controller="CentreConfiguration" asp-action="Index">Back to centre configuration</a>
+      </p>
     </div>
   </nav>
 }
 
 <div class="nhsuk-grid-row">
   <div class="nhsuk-grid-column-full">
     <h1 class="nhsuk-heading-xl">Manage delegate registration prompts</h1>
-  </div>
-</div>
 
-@foreach (var customField in Model.CustomFields) {
-  <div class="nhsuk-grid-row">
-    <div class="nhsuk-grid-column-full">
-      <details class="nhsuk-details nhsuk-expander">
-        <summary class="nhsuk-details__summary">
-          <div class="nhsuk-grid-row">
-            <div class="nhsuk-grid-column-three-quarters nhsuk-u-three-quarters-tablet">
-              <span class="nhsuk-details__summary-text">
-                @customField.CustomPromptName
-              </span>
-            </div>
-            <div class="nhsuk-grid-column-one-quarter nhsuk-u-one-quarter-tablet right-align-tag-column">
-              <span class="nhsuk-u-visually-hidden">This prompt is:</span>
-              <strong class="nhsuk-u-font-size-19 right-align-tag nhsuk-tag @(customField.Mandatory ? "" : "nhsuk-tag--grey")">
-                @(customField.Mandatory ? "Mandatory" : "Optional")
-              </strong>
-            </div>
-          </div>
-        </summary>
-        <div class="nhsuk-details__text">
-          <div class="nhsuk-grid-row nhsuk-u-margin-bottom-3">
-            <div class="nhsuk-grid-column-full">
-              @if (customField.Options.Count != 0) {
-                <p class="nhsuk-u-font-weight-bold">Answers delegate can choose from:</p>
-                <ul>
-                  @foreach (var answer in customField.Options)
-                  {
-                    <li class="word-break">@answer</li>
-                  }
-                </ul>
-              } else {
-                <p>This is a free text field. Delegates can input any text.</p>
-              }
-            </div>
-          </div>
-          <div class="nhsuk-grid-row">
-            <div class="nhsuk-grid-column-full">
-              <a class="nhsuk-button left-button-mobile-margin-bottom nhsuk-u-margin-right-2 nhsuk-u-margin-bottom-1" role="button" asp-controller="RegistrationPrompts" asp-action="EditRegistrationPromptStart" asp-route-promptNumber="@customField.CustomFieldId">
-                Edit
-              </a>
-              <a class="nhsuk-button delete-button nhsuk-button--secondary nhsuk-u-margin-bottom-1" role="button" asp-controller="RegistrationPrompts" asp-action="RemoveRegistrationPrompt" asp-route-promptNumber="@customField.CustomFieldId">
-                Remove
-              </a>
-            </div>
-          </div>
-        </div>
-      </details>
-    </div>
-  </div>
-}
+    <div class="@(hasSixPrompts ? "nhsuk-u-margin-bottom-7" : "nhsuk-u-margin-bottom-3")">

```

Comment:
What's special about 6 prompts?

Reply:
The Add new prompt button goes away. This margin + the card margins collapse, but this margin + the button margin don't collapse because the button has padding.

Reply:
Ah, okay. What about renaming `hasSixPrompts` to something like `canAddNewPrompt`? I'm guessing we don't let them put more than 6 prompts on the registration form? Or is it just that we only have 6 prompts that we can put on the form?

Reply:
There can never be more that 6 custom prompts defined for a centre (currently) as that is the number of columns in the database for the centre. (CustomPrompt?Id, F?Options, F?Mandatory columns). But yes, I think changing the name would be better.

Reply:
Ah, I guess it's actually `can'tAddNewPrompt`. But yeah.
I sort of feel like the logic `canAddNewPrompt = promptCount < 6` belongs somewhere a bit higher up as well, perhaps as a property on the model? Or maybe as a helper method? Because it's not page-specific - it's more than just `you're not allowed to add another prompt on this page`; it's `no centre should ever have more than 6 prompts`. But I'm not sure whether it would actually have any significant affect on future development. I'll pass you onto TL review anyway.
What do you think, @stellake?

Reply:
I personally don't think it is worth it. There are only really 2 places it is relevant, here to remove a button, and in the Add prompt save logic (if they somehow manage to get into there through our redirects and temp data service filters) where it is handled in a different way (if no empty slots to save into, don't save and go to error page).

Reply:
I think ideally, we'd have a "shouldDisplayAddNewPromptsButton" on the ViewModel as that's what the VCs are meant for, but I don't think it's too important and I'm inclined to leave it, because I think the current version of the code is readable and I'm inclined not to spend too long on this ticket. 👍